modules/dependencies.nix

```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.dependencies;
+
+  packages = {
+    inherit (pkgs)
+      curl
+      git
+      ;
+    ueberzug = pkgs.ueberzugpp;
+  };
+
+  mkDependencyOption = name: default: {
+    enable = lib.mkEnableOption "Add ${name} to dependencies.";
+
+    package = lib.mkPackageOption pkgs name { };
```

Comment:
`default` is unused.

---

IMO it is simpler to stick with string/list names/paths rather than trying to work with "real" packages, especially as that is how `mkPackageOption` is designed.

Additionally, do we anticipate a specific package warranting a bespoke `example`?

Maybe we need something like:

```nix
packages = {
  curl.default = ["curl"];
  git.default = ["git"];
  git.example = ["gitMinimal"];
  ueberzug.default = ["ueberzugpp"];
};

mkDependencyOption = name: args: {
  enable = lib.mkEnableOption "Install ${name} with nixvim. Its executables will be added to vim's `PATH`.";
  package = lib.mkPackageOption pkgs name args;
};
```

?